app/components/autocomplete/slash_suggestion/app_slash_suggestion/app_slash_suggestion.tsx

```diff
@@ -0,0 +1,238 @@
+import {
+    FlatList,
+    Platform,
+} from 'react-native';
+
+import {analytics} from '@init/analytics';
+import {Client4} from '@client/rest';
+import {isMinimumServerVersion} from '@mm-redux/utils/helpers';
+import {AutocompleteSuggestion} from '@mm-redux/types/integrations';
+import {Theme} from '@mm-redux/types/preferences';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+
+import SlashSuggestionItem from '../slash_suggestion_item';
+import {AppCommandParser} from '../app_command_parser/app_command_parser';
+import {COMMAND_SUGGESTION_CHANNEL, COMMAND_SUGGESTION_USER, ExtendedAutocompleteSuggestion} from '../app_command_parser/app_command_parser_dependencies';
+import AtMentionItem from '@components/autocomplete/at_mention_item';
+import ChannelMentionItem from '@components/autocomplete/channel_mention_item';
+
+export type Props = {
+    currentTeamId: string;
+    isSearch?: boolean;
+    maxListHeight?: number;
+    theme: Theme;
+    onChangeText: (text: string) => void;
+    onResultCountChange: (count: number) => void;
+    value: string;
+    nestedScrollEnabled?: boolean;
+    rootId?: string;
+    channelId: string;
+    appsEnabled: boolean;
+};
+
+type State = {
+    active: boolean;
+    dataSource: AutocompleteSuggestion[];
+}
+
+export default class SlashSuggestion extends PureComponent<Props, State> {
+    static defaultProps = {
+        defaultChannel: {},
+        value: '',
+    };
+
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
+    appCommandParser: AppCommandParser;
+
+    state = {
+        active: false,
+        dataSource: [],
+    };
+
+    constructor(props: Props, context: any) {
+        super(props);
+        this.appCommandParser = new AppCommandParser(null, context.intl, props.channelId, props.currentTeamId, props.rootId);
+    }
+
+    setActive(active: boolean) {
+        this.setState({active});
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        if (!isMinimumServerVersion(Client4.getServerVersion(), 5, 24) || !this.props.appsEnabled) {
+            this.setActive(false);
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        if (
+            (this.props.value === prevProps.value) ||
+            this.props.isSearch || this.props.value.startsWith('//') || !this.props.channelId
+        ) {
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        const {
+            value: nextValue,
+        } = this.props;
+
+        if (nextValue[0] !== '/') {
+            this.setActive(false);
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        if (nextValue.indexOf(' ') === -1) { // delegate base command to command parser
+            this.setActive(false);
+            this.props.onResultCountChange(0);
+            return;
+        }
+
+        if (!this.isAppCommand(nextValue, this.props.channelId, this.props.rootId)) {
```

Comment:
It looks like `this.isAppCommand` is expecting the team id as one of its arguments